node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -138,84 +136,57 @@ fn collect_externals_from_multi_config(
 }
 
 fn extract_values_vcl_fill_multiconfig_vec(
-    config_file_vcl: Box<dyn VirtualCommandLine>,
-    environment_vcl: Box<dyn VirtualCommandLine>,
-    commandline_vcl: Box<dyn VirtualCommandLine>,
-    user_specific_data: UserSpecifiedData,
+    multi_config_vec: Vec<Box<dyn VirtualCommandLine>>,
+    initialization_data: ConfigInitializationData,
+    app: &App,
 ) -> (Vec<String>, Vec<String>) {
+    let full_multi_config =
+        make_new_multi_config(app, multi_config_vec).expect("expected multi_config");
+    let config_file_specified = initialization_data.config_file.user_specified;
+    let config_file_path = initialization_data.config_file.item;
+    let check_value_from_mc = |value: Option<String>, var: &str, spec: bool| match value {
```

Comment:
Could `value` and `var` be better named? I don't understand the difference between them.

Or--and this is probably better--you could combine the last two parameters into one of your `FieldPair` types.